src/engine.cpp

```diff
@@ -53,42 +54,57 @@ const BlockSpecification &Engine::GetPhyloModelBlockSpecification() const {
 
 std::vector<double> Engine::LogLikelihoods(
     const UnrootedTreeCollection &tree_collection,
-    const EigenMatrixXdRef phylo_model_params, const bool rescaling) const {
+    const EigenMatrixXdRef phylo_model_params, const bool rescaling,
+    const std::optional<PhyloFlags> flags /*= std::nullopt*/) const {
   return FatBeagleParallelize<double, UnrootedTree, UnrootedTreeCollection>(
       FatBeagle::StaticUnrootedLogLikelihood, fat_beagles_, tree_collection,
-      phylo_model_params, rescaling);
+      phylo_model_params, rescaling, flags);
 }
 
-std::vector<double> Engine::LogLikelihoods(const RootedTreeCollection &tree_collection,
-                                           const EigenMatrixXdRef phylo_model_params,
-                                           const bool rescaling) const {
+std::vector<double> Engine::LogLikelihoods(
+    const RootedTreeCollection &tree_collection,
+    const EigenMatrixXdRef phylo_model_params, const bool rescaling,
+    const std::optional<PhyloFlags> flags /*= std::nullopt*/) const {
+  double (*func)(FatBeagle *, const RootedTree &, const std::optional<PhyloFlags>) =
```

Comment:
I'm curious why we need to do this sort of recasting.

Reply:
This is an artifact from when there were overloaded functions.  Removed.